src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => {
+    if (century) {
+      return person.sex === 'm' && Math.ceil(person.died / 100) === century;
+    } else {
+      return person.sex === 'm';
+    }
+  });
+  const age = men.map(person => person.died - person.born);
+  const averageAge = age.reduce((a, b) => (a + b), 0) / age.length;
+
+  return averageAge;
```

Comment:
You don't need to create a variable if you return its value right away.
```suggestion
  return age.reduce((a, b) => (a + b), 0) / age.length;
```